docs/markdown/using-snowchain/validating-and-relaying/relayer-flow.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+title: Relayer Flow
+parent: Validating and Relaying
+nav_order: 2
+---
+
+# Relayer Flow
+
+## Polkadot to Ethereum
+The relayer can be started in 2 different modes for relaying from Polkadot to Ethereum.
+
+## MMR-Epochs-Only mode
+This mode runs the relayer in a mode where it ensures that the Ethereum contract receives at least 1 MMR proof per epoch. It only interacts with the Relay Chain Light Client, not the Parachain Light Client.
+
+### Relayer process starts in MMR-Epochs-Only mode
+```mermaid!
+graph TD;
+  S1[Query Relay Chain Light Client for latest relayed MMR Epoch];
+  S1-->S2;
+  S2[Save as latestRelayedEpoch];
+  S2-->S3;
+  S2-->S5;
+  S3[Start querying/watching ethereum for new blocks];
+  S3-- New Ethereum Block Found -->S4;
+  S4[Add to Ethereum Block Processing Queue];
+  S5[Start querying/watching relay chain for new MMR Roots];
+  S5-- New Relay Chain MMR Root Found -->S6;
+  S6[Add to New MMR Root Processing Queue];
+```
+### Ethereum Block Processing Queue
+```mermaid!
+graph TD;
+  Eb1[New Block In Queue];
+  Eb1-- contains updated latestRelayedEpoch -->Eb2
+  Eb2[Save as latestRelayedEpoch];
+  Eb1-- no update -->Eb3
+  Eb3[Ignore];
+```
+### New MMR Root Processing Queue
+```mermaid!
+graph TD;
+  M1[New MMR Root In Queue];
+  M1-->M1a;
+  M1a[Is epoch == latestRelayedEpoch + 1?];
+  M1a-- No -->M2;
+  M1a-- Yes -->M3;
+  M2[Ignore];
+  M3[Is last block of epoch?];
+  M3-- Yes -->M4;
+  M4[Add new MMR Root to MMR Root Relay queue];
+  M3-- No -->M2;
```

Comment:
Again, let's use block height here, and process every MMR root for a higher block.

Reply:
same comment as above - this mode is only to ensure the bridge doesn't skip any epochs